Shared/Models/Subscription+CoreDataClass.swift

```diff
@@ -52,4 +52,21 @@ extension Subscription {
 
 extension Subscription : Identifiable {
     
+    func model() -> SubscriptionModel {
+        return .init(dbModel: self)
+    }
+    
+    func update(model: SubscriptionModel) {
+        let subType = SubscriptionType.allCases[Int.random(in: 0 ... SubscriptionType.allCases.count - 1)]
+        self.amount = model.amount
+        self.currency = model.currency.locale.identifier
+        self.cycle = "\(model.cycle.number.intValue)-\(model.cycle.period.rawValue)"
```

Comment:
are you really just planning to cycle and duration it as strings?
then casting them back to enums ?